Library/Homebrew/download_strategy.rb

```diff
@@ -312,6 +323,11 @@ def _fetch
     curl @url, "-C", downloaded_size, "-o", temporary_path
   end
 
+  def resolve_url(url)
+    m = Utils.popen_read("curl", "-I", url).match /^Location: (.+)$/
+    m.nil? ? url : resolve_url(m.captures.first)
```

Comment:
Sure, if we can find a reproducible example. I guess we could reintroduce ` --proto-redir ` but I'm not huge on introducing extra code unless there's a provable use case for it. A link that darts from secure => secure => insecure is a relative rarity.